apollo-router/src/warp_http_server_factory.rs

```diff
@@ -412,24 +414,25 @@ where
                   path: FullPath| {
                 let service = service.clone();
                 async move {
-                    let reply: Box<dyn Reply> = if accept.map(prefers_html).unwrap_or_default() {
-                        display_home_page()
-                    } else if let Ok(request) = graphql::Request::from_urlencoded_query(query) {
-                        run_graphql_request(
-                            service,
-                            authority,
-                            http::Method::GET,
-                            path,
-                            request,
-                            header_map,
-                        )
-                        .await
-                    } else {
-                        Box::new(warp::reply::with_status(
-                            "Invalid GraphQL request",
-                            StatusCode::BAD_REQUEST,
-                        ))
-                    };
+                    let reply: Box<dyn Reply> =
+                        if accept.map(prefers_html).unwrap_or_default() && display_landing_page {
```

Comment:
with this condition, if the browser sends a request while the landing page is disabled, it will try to do a graphql query instead of directly returning a "bad request" with "Invalid GraphQL request"

Reply:
It's directly catched by the plugin which check the presence of the query and then it returns this error:
```json
{
  "errors": [
     {
        "message": "Must provide query string.",
        "locations": [],
        "path": null
    }
  ]
}
```